service/udp.go

```diff
@@ -69,96 +69,59 @@ func findAccessKeyUDP(clientIP net.IP, dst, src []byte, cipherList CipherList) (
 	return nil, "", nil, errors.New("could not find valid cipher")
 }
 
-type udpService struct {
-	mu                sync.RWMutex // Protects .clientConn and .stopped
-	clientConn        net.PacketConn
-	stopped           bool
+type packetHandler struct {
 	natTimeout        time.Duration
 	ciphers           CipherList
 	m                 metrics.ShadowsocksMetrics
-	running           sync.WaitGroup
 	targetIPValidator onet.TargetIPValidator
 }
 
-// NewUDPService creates a UDPService
-func NewUDPService(natTimeout time.Duration, cipherList CipherList, m metrics.ShadowsocksMetrics) UDPService {
-	return &udpService{natTimeout: natTimeout, ciphers: cipherList, m: m, targetIPValidator: onet.RequirePublicIP}
+// NewPacketHandler creates a UDPService
+func NewPacketHandler(natTimeout time.Duration, cipherList CipherList, m metrics.ShadowsocksMetrics) PacketHandler {
+	return &packetHandler{natTimeout: natTimeout, ciphers: cipherList, m: m, targetIPValidator: onet.RequirePublicIP}
 }
 
-// UDPService is a running UDP shadowsocks proxy that can be stopped.
-type UDPService interface {
+// PacketHandler is a running UDP shadowsocks proxy that can be stopped.
+type PacketHandler interface {
 	// SetTargetIPValidator sets the function to be used to validate the target IP addresses.
 	SetTargetIPValidator(targetIPValidator onet.TargetIPValidator)
-	// Serve adopts the clientConn, and will not return until it is closed by Stop().
-	Serve(clientConn net.PacketConn) error
-	// Stop closes the clientConn and prevents further forwarding of packets.
-	Stop() error
-	// GracefulStop calls Stop(), and then blocks until all resources have been cleaned up.
-	GracefulStop() error
+	// Handle returns after clientConn closes and all the sub goroutines return.
+	Handle(clientConn net.PacketConn)
 }
 
-func (s *udpService) SetTargetIPValidator(targetIPValidator onet.TargetIPValidator) {
+func (s *packetHandler) SetTargetIPValidator(targetIPValidator onet.TargetIPValidator) {
```

Comment:
*nit* maybe use `p` (stands for `packetHandler`) instead of `s` (stands for `udpService`)?

Reply:
Replaced with `h`, for "handler"